spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -229,8 +229,8 @@ class SpdxCompoundExpression(
     override fun licenses() = left.licenses() + right.licenses()
 
     override fun disjunctiveNormalForm(): SpdxExpression {
-        val leftDnf = left.disjunctiveNormalForm()
-        val rightDnf = right.disjunctiveNormalForm()
+        // Both AND and OR operators are commutative, so it is fine to sort operands alphabetically.
+        val (leftDnf, rightDnf) = listOf(left, right).map { it.disjunctiveNormalForm() }.sortedBy { it.toString() }
```

Comment:
This is creating three temporary collections. How about this instead:
```
var leftDnf = left.disjunctiveNormalForm()
var rightDnf = right.disjunctiveNormalForm()

if (rightDnf.toString() < leftDnf.toString()) {
  leftDnf = rightDnf.also { rightDnf = leftDnf }
}
```

I was a bit surprised, but according to this it should work: https://stackoverflow.com/a/45377921/639421

Reply:
While I like this trick, the `var`s unfortunately break the smart-casts below.

Reply:
Then we should do it the traditional way with a placeholder variable, but I would prefer to have a bit more verbose code here than creating a lot of unnecessary collections.

Reply:
Sorry, but no. That's really becoming too ugly. As a compromise, I've now pushed a version that creates one temporary Pair. I hope that's fine for you.